common/src/main/java/org/ergoplatform/appkit/SelectTokensHelper.java

```diff
@@ -10,33 +10,60 @@
  */
 public class SelectTokensHelper {
     private final HashMap<String, Long> tokensLeft;
+    private boolean changeBoxNeeded;
 
     public SelectTokensHelper(Iterable<ErgoToken> tokensToSpent) {
         tokensLeft = new HashMap<>();
 
         for (ErgoToken ergoToken : tokensToSpent) {
             tokensLeft.put(ergoToken.getId().toString(), ergoToken.getValue());
         }
+
+        changeBoxNeeded = false;
     }
 
     /**
+     * Checks if the given tokens are needed to fulfill the tokens to
+     * spend
+     *
      * @return if the found tokens were needed to fill the tokens left
      */
-    public boolean foundNewTokens(Iterable<ErgoToken> foundTokens) {
+    public boolean areTokensNeeded(Iterable<ErgoToken> foundTokens) {
         boolean tokensNeeded = false;
         for (ErgoToken foundToken : foundTokens) {
             String tokenId = foundToken.getId().toString();
             if (tokensLeft.containsKey(tokenId)) {
                 Long currentValue = tokensLeft.get(tokenId);
                 if (currentValue > 0) {
                     tokensNeeded = true;
+                    break;
                 }
-                tokensLeft.put(tokenId, currentValue - foundToken.getValue());
             }
         }
         return tokensNeeded;
     }
 
+    /**
+     * Marks the given tokens as needed, subtracting the amount from the remaining amount of
+     * tokens to find.
+     * Also updates if a change box is needed
+     */
+    public void useTokens(Iterable<ErgoToken> foundTokens) {
```

Comment:
Please update the doc. It is not clear how tokens are marked. What is means "tokens to find"? Find where?